src/client.rs

```diff
@@ -1,39 +1,133 @@
 /// This is built into a separate binary called bynar-client
 //mod disk_manager;
 use std::fs::{read_to_string, File};
+use std::io::{Error, ErrorKind, Read, Write};
 use std::path::Path;
 use std::str::FromStr;
 
 //use disk_manager::disk_manager;
-use api::service::{Disk, OpOutcome};
+use api::service::{
+    Disk, Disks, JiraInfo, OpJiraTicketsResult, OpOutcome, OpOutcomeResult, ResultType,
+};
 use clap::{crate_authors, crate_version, App, Arg, ArgMatches, SubCommand};
-use helpers::error::BynarResult;
+use helpers::error::{BynarError, BynarResult};
 use hostname::get_hostname;
-use log::{error, info, trace};
+use log::{debug, error, info, trace};
 use simplelog::{CombinedLogger, Config, TermLogger, WriteLogger};
 use zmq::Socket;
+
+#[macro_use]
+mod util;
 /*
     CLI client to call functions over RPC
 */
 
-fn add_disk(s: &Socket, path: &Path, id: Option<u64>, simulate: bool) -> BynarResult<OpOutcome> {
-    let outcome = helpers::add_disk_request(s, path, id, simulate)?;
-    Ok(outcome)
+fn add_disk(
+    s: &Socket,
+    path: &Path,
+    id: Option<u64>,
+    client_id: Vec<u8>,
+    simulate: bool,
+) -> BynarResult<OpOutcomeResult> {
+    let mut sent = false;
+    //loop until socket is readable, then get the response
+    loop {
```

Comment:
This kinda looks like a spin lock.  If it is then you might want to insert a sleep here.  I don't know if `poll_events!` blocks

Reply:
I can't find any place in zmq get_sockopt or in the zmq crate that states that this blocks.  Also I'm fairly certain it doesn't, it basically just checks if there are any events in the socket and returns if there is or isn't (with like flags set for the specific event type if there are any)

Reply:
@mzhong1 What I think Chris is saying is that this loop just spins and busy waits until we've broken out of the loop.